### PR TITLE
Removing dependency on bluebird promises; putting response and body into error callback if available.

### PR DESCRIPTION
**Pull request by afrasso**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,5 +67,9 @@ For a full example inside an express app, see the [sample](sample) directory wit
 - Made resolveWithFullResponse the default
 
 ## 2.0
-- ES6 syntax, dropping node 5 support.
+- ES6 syntax, dropping Node.js 5 support.
 
+## 3.0
```
Comment (zmarois) [inline README.md]: It looks like you added a breaking change to the signature of the `RequestPromiseMiddlewareFramework` constructor. Can you list that? Personally, in major releases I like to have a big bold section outlining the breaking changes separate from the features.

Comment (afrasso) [reply README.md]: Can it be breaking and non-breaking at the same time? :) I will improve the changelog messaging, but note that the old signature for the constructor will still work.

File: README.md [hunk 2]
```diff
@@ -40,13 +40,21 @@ While the middleware does attempt to leave request-promise as pristine as possib
...
+const rpMiddlewareFramework = new RequestPromiseMiddlewareFramework({ rp: require("request-promise") }, middleware);
+const rp = rpMiddlewareFramework.getMiddlewareEnabledRequestPromise();
 ```
 
-You can then use returned ``rp`` object just like you normally would.
+Or, if you want to use an alternate Promise library such as bluebird, you can do so as follows:
+
+```
+const bluebirdPromise = require("bluebird");
+const rpMiddlewareFramework = new RequestPromiseMiddlewareFramework({ rp: require("request-promise"), P: bluebirdPromise }, middleware);
```
Comment (zmarois) [inline README.md]: `P` is not a great name for this override. I'll look at some common libraries and how they allow for it:

* [request-promise-any](https://github.com/request/request-promise-any) (maybe we should just use this. It is endorsed by [request-promise](https://github.com/request/request-promise)), which uses [any-promise](https://github.com/kevinbeaty/any-promise): field is named `Promise`
* [AWS SDK](https://aws.amazon.com/blogs/developer/support-for-promises-in-the-sdk/): `setPromiseDependency` (As an option, I'd infer it to `promiseDependency`)
* [mongoose](https://mongoosejs.com/docs/connections.html): `promiseLibrary`

File: README.md [hunk 3]
```diff
@@ -40,13 +40,21 @@ While the middleware does attempt to leave request-promise as pristine as possib
...
+Once you've defined your middleware, you can simply register it by creating a new framework object, and then getting the request object. If you want to use native Promises, you can do so as follows:
 ```
-var rpMiddlewareFramework = new RequestPromiseMiddlewareFramework(require("request-promise"), middleware);
-var rp = rpMiddlewareFramework.getMiddlewareEnabledRequestPromise();
+const rpMiddlewareFramework = new RequestPromiseMiddlewareFramework({ rp: require("request-promise") }, middleware);
+const rp = rpMiddlewareFramework.getMiddlewareEnabledRequestPromise();
 ```
 
-You can then use returned ``rp`` object just like you normally would.
+Or, if you want to use an alternate Promise library such as bluebird, you can do so as follows:
```
Comment (zmarois) [inline README.md]: Same comment about how we document parameters:

I'd love JSDoc on the class to describe it's parameters. Lacking that, just a list of parameters and what they do is nice.

I don't find a slew of examples of how to use each parameter easy to grok,

File: README.md (hunk 1 above)
Comment (zmarois) [reply README.md]: If the old signature still works, then it isn't breaking. (Dropping node6 support kinda is, but I'm not sure I see other libraries versioning for that purpose).

File: README.md (hunk 2 above)
Comment (afrasso) [reply README.md]: The `any-promise` stuff is intriguing although it replaces one required dependency (bluebird) with another (any-promise), and it requires users of the library to know to register a Promise dependency with any-promise (which they might not do). With the implementation I used at least it's deterministic which Promise library is chosen. That said, I switched from P to PromiseDependency.

File: README.md (hunk 3 above)
Comment (afrasso) [reply README.md]: OK, I added the JSDocs.

**Pull request merged**